pdc/apps/common/viewsets.py

```diff
@@ -63,6 +65,64 @@ def perform_create(self, serializer):
                                        _dumps_json(item.export()))
 
 
+def get_related_attr(obj, spec):
+    fields = spec.split('__')
+    for f in fields:
+        obj = getattr(obj, f)
+    return obj
+
+
+class NotificationMixin(object):
+    def get_object(self):
+        obj = super(NotificationMixin, self).get_object()
+        serializer_class = self.get_serializer_class()
+        self.initial_status = serializer_class(instance=obj,
+                                               context={'request': self.request}).data
+        return obj
+
+    def _get_url(self, basename):
+        if not hasattr(self, 'object') or not self.object:
+            return None
+        if hasattr(self, 'lookup_fields'):
+            lookup_field_values = [get_related_attr(self.object, f) for f, _ in self.lookup_fields]
+            lookup_field = '/'.join(lookup_field_values)
+        else:
+            lookup_field = getattr(self.object, self.lookup_field)
+        rel_url = reverse('%s-detail' % basename, args=[lookup_field])
+        return self.request.build_absolute_uri(rel_url)
+
+    def _send_message(self, action, msg):
+        for resource, cls, basename in router.registry:
+            if isinstance(self, cls):
+                break
+        else:
+            raise RuntimeError('What do we do now? Is it even possible?')
```

Comment:
I would suspect that if this happens there is no route to the view and therefore it must have been changed internally.

Reply:
Would an internal change be coming via the viewset?

Reply:
I think this probably cannot happen, so keeping an exception here is fine.